OpenRA.Mods.AS/Traits/NydusPrimaryExit.cs

```diff
@@ -0,0 +1,104 @@
+﻿#region Copyright & License Information
+/*
+ * Modded by Boolbada of OP mod, from Primary production faciility designation logic.
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+/* Works without base engine modification */
+namespace OpenRA.Mods.AS.Traits
+{
+	static class NydusPrimaryExitExts
+	{
+		public static bool IsPrimaryNydusExit(this Actor a)
+		{
+			var pb = a.TraitOrDefault<NydusPrimaryExit>();
+			return pb != null && pb.IsPrimary;
+		}
+	}
+
+	[Desc("Used with Nydus trait for primary exit designation")]
+	public class NydusPrimaryExitInfo : ITraitInfo
```

Comment:
Come to think of it...

If you do the Actor,Trait cache within the player trait, this trait should be merged into the main trait. That would make things more easier.